### PR TITLE
Delete STUBLINKER_GENERATES_UNWIND_INFO

### PR DESCRIPTION
**Pull request by jkotas**
None of the stublinker stubs need this anymore

### CODE REVIEW

**Comment by jkotas** [comment]
> @jkotas Is there any impact to R2R for x86?

This change is no-op for x86. The stub unwind info was not emitted on x86 before this change (both Windows and non-Windows).

File: src/coreclr/vm/riscv64/stubs.cpp [hunk 1]
```diff
@@ -1104,117 +1104,6 @@ void StubLinkerCPU::EmitMovConstant(IntReg reg, UINT64 imm)
...
-    // Besides, saving argument registers cannot be expressed by the unwind code encodings.
-    // So, we'll push saving the argument registers to the very last in the prolog, skip restoring it in epilog,
-    // and also skip reporting it to the OS.
-    //
-    // Another bit that we can save is resetting the frame pointer.
-    // This is not necessary when the SP doesn't get modified beyond prolog and epilog. (i.e no alloca/localloc)
-    // And in that case we don't need to report setting up the FP either.
-
-    // 1. Relocate SP
-    EmitSubImm(RegSp, RegSp, totalPaddedFrameSize);
```
Comment (tomeksowi) [inline src/coreclr/vm/riscv64/stubs.cpp:1170]: After #107282 that was the last call site, you can remove EmitSubImm from RISC-V.

File: src/coreclr/vm/riscv64/stubs.cpp [hunk 2]
```diff
@@ -1104,117 +1104,6 @@ void StubLinkerCPU::EmitMovConstant(IntReg reg, UINT64 imm)
...
-    EmitStore(RegRa, RegSp, cbStackSpace + sizeof(void*));
-
-    // 3. Set the frame pointer to the Canonical Frame Address or CFA, which is the stack pointer value on entry to the
-    // current procedure
-    EmitAddImm(RegFp, RegSp, totalPaddedFrameSize);
-
-    // 4. Store floating point argument registers
-    _ASSERTE(cFpRegArgs <= 8);
-    for (unsigned short i = 0; i < cFpRegArgs; i++)
-        EmitStore(FloatReg(i + 10), RegSp, cbOffset + i * sizeof(void*));
```
Comment (tomeksowi) [inline src/coreclr/vm/riscv64/stubs.cpp:1185]: Same here, you can remove FloatReg overloads of Emit(Load|Store) and FloatReg itself.

File: src/coreclr/vm/riscv64/stubs.cpp (hunk 1 above)
Comment (jkotas) [reply src/coreclr/vm/riscv64/stubs.cpp:1170]: Deleted. Thanks!

**Review by janvorli — APPROVED** [review]
LGTM, thank you!

**Pull request merged**